server/monitor/subscriptions.go

```diff
@@ -205,41 +219,43 @@ func (m *Monitor) recreateChannelSubscription(subscriptionID, teamID, channelID,
 
 	m.metrics.ObserveSubscription(metrics.SubscriptionReconnected)
 
-	if subscriptionID != "" {
-		if err = m.store.DeleteSubscription(subscriptionID); err != nil {
-			m.api.LogDebug("Unable to delete old channel subscription from DB", "subscriptionID", subscriptionID, "error", err.Error())
+	if subscriptionID == "" {
+		tx, err := m.store.BeginTx()
```

Comment:
Isn't this adding to the transaction logic outside the store?  I thought we were going to undo that.

Reply:
Ah, I see this is just continuing with the existing code.  I guess a different PR will fix it.

Reply:
@jespino, would we consider fixing this as part of this PR? I'm finding it hard to understand the net effect of these changes by virtue of having to preserve the transaction juggling outside the store.

Reply:
This is gone after the transactions exposure is removed from the API